operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -0,0 +1,309 @@
+        try (InputStream in = getClass().getClassLoader().getResourceAsStream(OPERATION_DEFINITION_FILE)) {
+            return FHIRParser.parser(Format.JSON).parse(in);
+        } catch (FHIRParserException e) {
+            throw new Error("There has been a parsing error in the operation definition of the $everything operation.", e);
+        } catch (IOException e) {
+            throw new Error("There has been an IO error reading the operation definition of the $everything operation.", e);
+        }
+    }
+
+    @Override
+    protected Parameters doInvoke(FHIROperationContext operationContext, Class<? extends Resource> resourceType, String logicalId, String versionId, Parameters parameters, FHIRResourceHelpers resourceHelper) throws FHIROperationException {
+        LOG.entering(this.getClass().getName(), "doInvoke");
+
+        Patient patient = null;
+        try {
+            patient = (Patient) resourceHelper.doRead(PATIENT, logicalId, false, false, null, null);
+        } catch (Exception e) {
+            FHIROperationException exceptionWithIssue = buildExceptionWithIssue("An unexpected error occurred while reading patient " + logicalId, IssueType.EXCEPTION);
+            LOG.throwing(this.getClass().getName(), "doInvoke", exceptionWithIssue);
+            throw exceptionWithIssue;
+        }
+        if (patient == null) {
+            FHIROperationException exceptionWithIssue = buildExceptionWithIssue("Patient with ID " + logicalId + " does not exist.", IssueType.NOT_FOUND);
+            LOG.throwing(this.getClass().getName(), "doInvoke", exceptionWithIssue);
+            throw exceptionWithIssue;
+        }
+
+        Entry patientEntry = buildPatientEntry(operationContext, patient);
+        List<Entry> allEntries = new ArrayList<>(SearchConstants.MAX_PAGE_SIZE);
+        allEntries.add(patientEntry);
+        
+        // We can't always use the "date" query parameter to query by clinical date, only with some resources.
+        // Initial list obtained from the github issue: https://github.com/IBM/FHIR/issues/1044#issuecomment-769788097
+        // Otherwise the search throws an exception. We create a params map with and without and use as needed
+        MultivaluedMap<String, String> queryParameters = parseQueryParameters(parameters);
+        MultivaluedMap<String, String> queryParametersWithoutDates = new MultivaluedHashMap<String, String>(queryParameters);
+        queryParametersWithoutDates.remove(DATE_QUERY_PARAMETER);
+        
+        List<String> resourceTypesOverride = getOverridenIncludedResourceTypes(parameters);
+        List<String> resourceTypes = resourceTypesOverride.isEmpty() ? defaultResourceTypes : resourceTypesOverride;
+        
+        for (String compartmentType : resourceTypes) {
+            MultivaluedMap<String, String> searchParameters = queryParameters;
+            if (!SUPPORT_CLINICAL_DATE_QUERY.contains(compartmentType)) {
+                searchParameters = queryParametersWithoutDates;
```

Comment:
Would be nice to add a log msg here if the request specified the date parameter but it's not valid for this resource type.

Reply:
Sure done.